### PR TITLE
authn: save attributes to RequestInfo.metadata for later use in RBAC filter.

### PR DESCRIPTION
**Pull request by yangminzhu**
Support to pass the auth attributes in metadata for later use in RBAC filter.

See istio/istio#6377

Signed-off-by: Yangmin Zhu <ymzhu@google.com>

### CODE REVIEW

File: src/istio/control/http/attributes_builder.cc [hunk 1]
```diff
@@ -62,6 +68,44 @@ void AttributesBuilder::ExtractRequestHeaderAttributes(CheckData *check_data) {
   }
 }
 
+void AttributesBuilder::SaveAuthAttributesToStruct(
```
Comment (qiwzhang) [inline src/istio/control/http/attributes_builder.cc]: It is better to move this function to https://github.com/istio/proxy/blob/master/src/envoy/utils/authn.h#L29

Comment (yangminzhu) [reply src/istio/control/http/attributes_builder.cc]: I tried that but it couldn't access the attributes names (See [here](https://github.com/istio/proxy/blob/624540603b7ebb206335d5cb375faa4d8ffcfbdc/src/istio/control/BUILD#L28)), I don't want to change the visibility of those attributes names so I added this in attributes_builder.cc, also it's closer to the `ExtractAuthAttributes()` function that makes it clear we are setting the right names.

Comment (qiwzhang) [reply src/istio/control/http/attributes_builder.cc]: I see. Then I like to have following changes to expose attribute_names

1) move attribute_name.h to include/istio/utils/
2) create a separate library as attribute_names in src/istio/util  and set its visibility as public

Comment (yangminzhu) [reply src/istio/control/http/attributes_builder.cc]: Done.

File: include/istio/utils/BUILD [hunk 2]
```diff
@@ -34,3 +34,11 @@ cc_library(
     ],
     visibility = ["//visibility:public"],
 )
+
+cc_library(
+    name = "attribute_names_lib",
```
Comment (qiwzhang) [inline include/istio/utils/BUILD]: attribute_names_header

File: src/istio/control/http/request_handler_impl_test.cc [hunk 3]
```diff
@@ -403,8 +403,9 @@ TEST_F(RequestHandlerImplTest, TestDefaultApiKey) {
                           TransportCheckFunc transport,
                           CheckDoneFunc on_done) -> CancelFunc {
         auto map = attributes.attributes();
-        EXPECT_EQ(map[AttributeName::kRequestApiKey].string_value(),
-                  "test-api-key");
+        EXPECT_EQ(
+            map[istio::utils::AttributeName::kRequestApiKey].string_value(),
```
Comment (qiwzhang) [inline src/istio/control/http/request_handler_impl_test.cc]: nit:  if a module is under istio namaespace,  you can remove namespace prefix "istio::"

File: include/istio/utils/BUILD (hunk 2 above)
Comment (yangminzhu) [reply include/istio/utils/BUILD]: Done.

File: src/istio/control/http/request_handler_impl_test.cc (hunk 3 above)
Comment (yangminzhu) [reply src/istio/control/http/request_handler_impl_test.cc]: Done.

**Pull request merged**